shell/testing/tester_main.cc

```diff
@@ -44,25 +44,28 @@ class ScriptCompletionTaskObserver {
   }
 
   void DidProcessTask() {
-    last_error_ = shell_.GetUIIsolateLastError();
-    if (shell_.EngineHasLivePorts()) {
-      // The UI isolate still has live ports and is running. Nothing to do
-      // just yet.
-      return;
-    }
-
-    if (run_forever_) {
-      // We need this script to run forever. We have already recorded the last
-      // error. Keep going.
-      return;
-    }
-
-    if (!has_terminated) {
-      // Only try to terminate the loop once.
-      has_terminated = true;
-      main_task_runner_->PostTask(
-          []() { fml::MessageLoop::GetCurrent().Terminate(); });
-    }
+    // We are installed on the UI thread so we need to move to the platform task
+    // runner
+    fml::TaskRunner::RunNowOrPostTask(main_task_runner_, [this]() {
+      last_error_ = shell_.GetUIIsolateLastError();
+      if (shell_.EngineHasLivePorts()) {
+        // The UI isolate still has live ports and is running. Nothing to do
+        // just yet.
+        return;
+      }
+
+      if (run_forever_) {
+        // We need this script to run forever. We have already recorded the last
+        // error. Keep going.
+        return;
+      }
+
+      if (!has_terminated) {
+        // Only try to terminate the loop once.
+        has_terminated = true;
+        fml::MessageLoop::GetCurrent().Terminate();
```

Comment:
Expanding on the earlier comment, this call needs to happen on the platform task runner (by posting to it).